packages/material-ui/src/ToggleButton/ToggleButton.js

```diff
@@ -36,91 +35,58 @@ const ToggleButtonRoot = styled(ButtonBase, {
 
     return [styles.root, styles[`size${capitalize(styleProps.size)}`]];
   },
-})(({ theme, styleProps }) => ({
-  /* Styles applied to the root element. */
-  ...theme.typography.button,
-  borderRadius: theme.shape.borderRadius,
-  padding: 11,
-  border: `1px solid ${theme.palette.divider}`,
-  color: theme.palette.action.active,
-  /* Styles applied to the root element if `fullWidth={true}`. */
-  ...(styleProps.fullWidth && {
-    width: '100%',
-  }),
-  [`&.${toggleButtonClasses.disabled}`]: {
-    color: theme.palette.action.disabled,
-    border: `1px solid ${theme.palette.action.disabledBackground}`,
-  },
-  '&:hover': {
-    textDecoration: 'none',
-    // Reset on mouse devices
-    backgroundColor: alpha(theme.palette.text.primary, theme.palette.action.hoverOpacity),
-    '@media (hover: none)': {
-      backgroundColor: 'transparent',
+})(({ theme, styleProps }) => {
+  const selectedColor =
+    styleProps.color === 'standard'
+      ? theme.palette.text.primary
+      : theme.palette[styleProps.color].main;
+  return {
+    ...theme.typography.button,
+    borderRadius: theme.shape.borderRadius,
+    padding: 11,
+    border: `1px solid ${theme.palette.divider}`,
+    color: theme.palette.action.active,
+    /* Styles applied to the root element if `fullWidth={true}`. */
+    ...(styleProps.fullWidth && {
+      width: '100%',
+    }),
+    [`&.${toggleButtonClasses.disabled}`]: {
+      color: theme.palette.action.disabled,
+      border: `1px solid ${theme.palette.action.disabledBackground}`,
```

Comment:
This change looks hard to review, may be this explanation helps. 
The change is to remove the duplicate code and combine
- ...(styleProps.color === 'standard' &&
- ...(styleProps.color !== 'standard' &&
into one scope with variable `selectedColor` (line 39) and that's all.

Reply:
Hiding whitespace changes helped me a lot here.